packages/cache/src/internal/cacheUtils.ts

```diff
@@ -91,7 +91,7 @@ export async function getCompressionMethod(): Promise<CompressionMethod> {
   }
 
   const versionOutput = await getVersion('zstd')
-  const version = semver.clean(versionOutput)
+  const version = semver.clean(versionOutput.match(/v\d\.\d\.\d/)[0])
```

Comment:
This doesn't match if the major or minor version have more than one digit, and it only keeps the most significant digit of the patch version.

```suggestion
  const version = semver.clean(versionOutput.match(/v\d+\.\d+\.\d+/)[0])
```